File: modules/Plugin/src/daybreak/abilitywar/utils/base/concurrent/TimeUnit.java
package daybreak.abilitywar.utils.base.concurrent;

public enum TimeUnit {
	TICKS(1),
	SECONDS(20),
	MINUTES(1200);

	private final int inTicks;

	TimeUnit(final int inTicks) {
		this.inTicks = inTicks;
	}

	public int toTicks(final int duration) {
		return duration * this.inTicks;
	}

}

File: modules/Plugin/src/daybreak/abilitywar/utils/base/concurrent/TimeUnit.kt
package daybreak.abilitywar.utils.base.concurrent

enum class TimeUnit(private val inTicks: Int) {
	TICKS(1),
	SECONDS(20),
	MINUTES(1200);

	fun toTicks(duration: Int): Int {
		return duration * this.inTicks
	}

}